src/libraries/System.Net.Security/src/System/Net/CertificateValidationPal.OSX.cs

```diff
@@ -71,15 +71,15 @@ internal static partial class CertificateValidationPal
             {
                 long chainSize = Interop.AppleCrypto.X509ChainGetChainSize(chainHandle);
 
-                if (retrieveChainCertificates)
+                if (retrieveChainCertificates && chainSize > 1)
                 {
                     chain ??= new X509Chain();
                     if (chainPolicy != null)
                     {
                         chain.ChainPolicy = chainPolicy;
                     }
 
-                    for (int i = 0; i < chainSize; i++)
+                    for (int i = 1; i < chainSize; i++)
```

Comment:
It will not add leaf certificate to ExtraStore (since it is returned separately) to make it more consistent with other platforms.

Reply:
Can you add a comment about why it skips the first item?

Reply:
updated.